crabs/detection_tracking/inference_model.py

```diff
@@ -0,0 +1,394 @@
+        ) / 2
+
+    def compute_motion_frame(
+        self,
+        frame_delta,
+        background_subtracted_frame,
+        mean_blurred_frame,
+        max_abs_blurred_frame,
+    ):
+        """
+        _summary_.
+
+        Parameters
+        ----------
+        frame_delta : int
+            difference in number of frames used to compute the motion
+            channel is computed
+        background_subtracted_frame : np.array
+            normalised difference between the blurred frame f and
+            the mean blurred frame
+        mean_blurred_frame : np.array
+            mean of all blurred frames in the video
+        max_abs_blurred_frame : np.array
+            pixelwise max absolute value across all blurred frames in the video
+
+        Returns
+        -------
+        motion_frame : np.array
+            absolute difference between the background subtracted frame f
+            and the background subtracted frame f+delta
+        """
+        # compute the blurred frame frame_idx+delta
+        _, blurred_frame_delta = self.apply_grayscale_and_blur(
+            frame_delta,
+            [5, 5],
+            0,
+        )
+        # compute the background subtracted for frame_idx + delta
+        background_subtracted_frame_delta = (
+            self.compute_background_subtracted_frame(
+                blurred_frame_delta,
+                mean_blurred_frame,
+                max_abs_blurred_frame,
+            )
+        )
+
+        # compute the motion channel for frame_idx
+        return np.abs(
+            background_subtracted_frame_delta - background_subtracted_frame,
+        )
+
+    def __inference(
+        self, final_frame: np.ndarray, frame: np.ndarray
+    ) -> np.ndarray:
+        """
+        Perform inference on a single frame of the video.
+
+        Args:
+            frame (np.ndarray): The input frame as a NumPy array.
+
+        Returns:
+            None
+        """
+        self.trained_model.eval()
+        transform = transforms.Compose(
+            [
+                transforms.ToTensor(),
+            ]
+        )
+        img = transform(final_frame)
+        img = img.to(device)
+
+        img = img.unsqueeze(0)
+        prediction = self.trained_model(img)
+        pred_score = list(prediction[0]["scores"].detach().cpu().numpy())
+
+        if not self.args.sort:
+            from _inference import inference_detection
+
+            frame_out = inference_detection(
+                frame, prediction, pred_score, self.score_threshold
+            )
+
+        else:
+            from _inference import inference_tracking
+
+            frame_out = inference_tracking(
+                frame,
+                prediction,
+                pred_score,
+                self.score_threshold,
+                self.sort_crab,
+            )
+        return frame_out
+
+    def _load_video(self) -> None:
+        """
+        Load the input video and perform inference on its frames.
```

Comment:
maybe also include in the docstring that the multi channels are extracted if required.